### PR TITLE
Feature Branch for v2.4.6

### PR DESCRIPTION
**Pull request by attackant**
### Bugs Fixed

- #1057 - Resolved type error on Automation admin screen
- #1056 - Resolved error fetching list of sections 
- #1055 - Moved async functions outside useEffect to avoid synchronous errors
- #1051 - Updated npm dependencies to match WordPress
- #1050 - Resolved uncaught promise error when saving automation rules
- #1048 - Fix posting bug caused by invalid document name. Thanks @alamedapost for your help!
- #1030 - Fixed render error with "publish-to-apple-news" plugin. Thanks @interactivebytes for reporting!
- #1028 - Fixed inability to select checklist options in Post Apple News Options plugin. Thanks @aeroplanejane for reporting!
- #1018 - Fixed bug deleting automation rules which removed all page contents. Thanks @jtroynousky for reporting!
- #1014 - Resolved invalid document exception while processing articles. Thanks @misaalami for your help!  

### Enhancements 

- #1024 - Upgraded to Node.js 20 and npm 10

### Documentation

- #1042 - Added info for using plugin with News Preview app. Thanks @benlk for the documentation!

### CODE REVIEW

File: admin/apple-actions/class-action.php [hunk 1]
```diff
@@ -48,6 +48,6 @@ abstract public function perform();
 	 * @return mixed The value for the setting.
 	 */
 	protected function get_setting( $name ) {
-		return $this->settings->get( $name );
+		return $this->settings->__get( $name );
```
Comment (attackant) [inline admin/apple-actions/class-action.php:51]: `get` is deprecated in favor of `__get`

File: admin/class-admin-apple-themes.php [hunk 2]
```diff
@@ -574,7 +574,9 @@ public function theme_nag() {
 				__( 'It looks like you are using the default theme. You can choose a new theme or customize your theme on the %1$sthemes page%2$s.', 'apple-news' ),
 				'<a href="' . esc_url( admin_url( 'admin.php?page=apple-news-themes' ) ) . '">',
 				'</a>'
-			)
+			),
+			get_current_user_id(),
+			true
 		);
```
Comment (attackant) [inline admin/class-admin-apple-themes.php:580]: Makes notice dismissable

File: package.json [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
   "name": "publish-to-apple-news",
-  "version": "2.4.5",
+  "version": "2.4.6",
   "license": "GPLv3",
   "main": "index.php",
   "engines": {
-    "node": "16",
-    "npm": "8"
+    "node": ">=18.0.0 <22.0.0",
+    "npm": ">=9.0.0 <11.0.0"
```
Comment (mslinnea) [inline package.json]: Is a range necessary?  We had a problem on another project caused by a range, as some packages may load different versions.

File: readme.txt [hunk 4]
```diff
@@ -46,6 +46,21 @@ Please visit our [wiki](https://github.com/alleyinteractive/apple-news/wiki) for
...
+* Bugfix: #1056 - Resolved error fetching list of sections
+* Bugfix: #1055 - Moved async functions outside useEffect to avoid synchronous errors
+* Bugfix: #1051 - Updated npm dependencies to match WordPress
+* Bugfix: #1050 - Resolved uncaught promise error when saving automation rules
+* Bugfix: #1048 - Fix posting bug caused by invalid document name.
+* Bugfix: #1030 - Fixed render error with "publish-to-apple-news" plugin.
+* Bugfix: #1028 - Fixed inability to select checklist options in Post Apple News Options plugin.
+* Bugfix: #1018 - Fixed bug deleting automation rules which removed all page contents.
+* Bugfix: #1014 - Resolved invalid document exception while processing articles.
+* Enhancement: Upgraded to Node.js 20 and npm 10.
```
Comment (mslinnea) [inline readme.txt:61]: Could the engines be set to these specifically?

**Review by mslinnea — APPROVED** [review]
Approving but consider that specific node and npm versions may prevent issues.

File: package.json (hunk 3 above)
Comment (attackant) [reply package.json]: Thanks Linnea, I just made a change to tighten down the ranges, but I think keeping them is a good idea in order to include point releases in within the one version we are on now, in this case 20. Please give the changes a re-review with you thoughts. Thanks!

File: readme.txt (hunk 4 above)
Comment (attackant) [reply readme.txt:61]: See my last commit, I restricted to a major version but still allowing point releases in the range.

File: package.json (hunk 3 above)
Comment (attackant) [reply package.json]: Removed the range

File: readme.txt (hunk 4 above)
Comment (attackant) [reply readme.txt:61]: Done

**Review by mslinnea — APPROVED** [review]
Looks good! 🐴

**Pull request merged**